s3tests/common.py

```diff
@@ -53,6 +53,29 @@ def nuke_bucket(bucket):
                 key.set_canned_acl('private')
                 key.delete()
                 deleted_cnt += 1
+        # Delete old versioned keys
+        deleted_cnt = 1
+        while deleted_cnt:
+            deleted_cnt = 0
+            for key in bucket.list_versions():
+                print 'Cleaning bucket {bucket} key {key} version {verion}'.format(
+                    bucket=bucket,
+                    key=key,
+                    version=key.version_id,
+                    )
+                key.set_canned_acl('private')
```

Comment:
@robbat2 moreover, why not modifying the regular object removal above to do versioned object listing, so that it removes all objects, instead of doing it in two paths?